crates/providers/kv-redis/src/bin/kvredis.rs

```diff
@@ -346,35 +319,186 @@ impl KvRedisProvider {
     /// with redis operations, but any control commands for new actor links
     /// or removal of actor links may need to wait for in-progress operations to complete.
     /// That should be rare, because most links are passed to the provider at startup.
-    async fn exec<T: FromRedisValue>(&self, ctx: &Context, cmd: &mut redis::Cmd) -> RpcResult<T> {
+    async fn exec<T: FromRedisValue>(
+        &self,
+        ctx: &Context,
+        cmd: &mut redis::Cmd,
+    ) -> Result<T, String> {
         let actor_id = ctx
             .actor
             .as_ref()
-            .ok_or_else(|| RpcError::InvalidParameter("no actor in request".to_string()))?;
+            .ok_or_else(|| "no actor in request".to_string())?;
         // get read lock on actor-connections hashmap
         let rd = self.actors.read().await;
         let rc = rd
             .get(actor_id)
-            .ok_or_else(|| RpcError::InvalidParameter(format!("No Redis connection found for {}. Please ensure the URL supplied in the link definition is a valid Redis URL", actor_id)))?;
+            .ok_or_else(||format!("No Redis connection found for {}. Please ensure the URL supplied in the link definition is a valid Redis URL", actor_id))?;
         // get write lock on this actor's connection
         let mut con = rc.write().await;
-        cmd.query_async(con.deref_mut()).await.map_err(to_rpc_err)
+        cmd.query_async(con.deref_mut())
+            .await
+            .map_err(|e| e.to_string())
     }
 }
 
-fn get_redis_url(link_values: &HashMap<String, String>, default_connect_url: &str) -> String {
+fn get_redis_url(link_values: &[(String, String)], default_connect_url: &str) -> String {
     link_values
         .iter()
         .find(|(key, _value)| key.eq_ignore_ascii_case(REDIS_URL_KEY))
         .map(|(_key, url)| url.to_owned())
         .unwrap_or_else(|| default_connect_url.to_owned())
 }
 
+#[async_trait::async_trait]
+impl wasmcloud_provider_sdk::MessageDispatch for KvRedisProvider {
```

Comment:
Does dispatch belong here? Not in the generated interface file?

Reply:
Yeah because generated interface in this case is in a separate "crate" (the lib part) than the actual provider impl and struct